java7/lib/src/main/java/com/truelayer/signing/Utils.java

```diff
@@ -0,0 +1,55 @@
+package com.truelayer.signing;
+
+import com.nimbusds.jose.util.Base64URL;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Utils {
+
+    protected static Map<String, Object> jwsHeaderMap(String kid, Map<HeaderName, String> headers) {
+        StringBuilder sb = new StringBuilder();
+        int counter = 0;
+        for (Map.Entry<HeaderName, String> entry : headers.entrySet()) {
+            HeaderName name = entry.getKey();
+            sb.append(name.getName());
+            if (counter < headers.size() - 1)
+                sb.append(",");
+            counter++;
+        }
+
+        Map<String, Object> jwsheader = new HashMap<>();
+        jwsheader.put("alg", "ES512");
+        jwsheader.put("kid", kid);
+        jwsheader.put("tl_version", "2");
+        jwsheader.put("tl_headers", sb.toString());
+        return jwsheader;
+    }
+
+    public static Base64URL buildPayload(Map<HeaderName, String> headers, String method, String path, byte[] body) {
+
+        StringBuilder headerStringBuilder = new StringBuilder();
+        for (Map.Entry<HeaderName, String> entry : headers.entrySet()) {
+            HeaderName name = entry.getKey();
+            String val = entry.getValue();
+            headerStringBuilder.append(name.getName());
```

Comment:
is there a case where name can be null here?

Reply:
I dont think it is possible, as existence of a header with a name is unlikely